File: jdt-dpgen-cli/src/main/resources/dpgen-output/src/main/scala/com/gof/behavioral/visitor/ConcreteElementA.scala
package com.gof.behavioral.visitor

//remove if not needed
import scala.collection.JavaConversions._

class ConcreteElementA extends Element {

  var counter: Int = _

  def operationElementA() {
    counter += 1
  }

  protected def getcounter(): Int = counter

  override def accept(visitor: Visitor) {
    visitor.visitConcreteElementA(this)
  }
}

File: jdt-dpgen-cli/src/main/resources/dpgen-output/src/main/java/com/gof/behavioral/visitor/ConcreteElementA.java
package com.gof.behavioral.visitor;
public class ConcreteElementA implements Element {

	public int counter;

	public void operationElementA() {
		counter++;
	}

	protected int getcounter() {
		return counter;
	}

	@Override
	public void accept(Visitor visitor) {
		visitor.visitConcreteElementA(this);
	}
}